### PR TITLE
Make CW popup showing on hover and IntelliSense popup non-conflicting

### PR DESCRIPTION
**Pull request by andrewyuq**
1. CW popup is hidden by default, on hovering over the suggestion text it will show.
2. On hovering over the suggestion, IntelliSense popup(if it's showing) will increase it's alpha to 0.8
3. Upon typing, make CW hidden and IntelliSense not transparent.

https://github.com/user-attachments/assets/367c5c2e-5132-4230-a712-6dc2cd5149a9

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/model/CodeWhispererModel.kt [hunk 1]
```diff
@@ -130,6 +134,16 @@ data class SessionContext(
     var hasAccepted: Boolean = false
 ) : Disposable {
     private var isDisposed = false
+    init {
+        project.messageBus.connect().subscribe(
```
Comment (Will-ShaoHua) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/model/CodeWhispererModel.kt:138]: should we pass `this` as the disposable parent?

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt [hunk 2]
```diff
@@ -894,3 +902,7 @@ data class ResponseContext(
 interface CodeWhispererCodeCompletionServiceListener {
     fun onSuccess(fileContextInfo: FileContextInfo) {}
 }
+
+interface CodeWhispererIntelliSenseOnHoverListener {
+    fun onEnter() {}
```
Comment (Will-ShaoHua) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:907]: do we need this default implementation or simply make it abstract?

**Review by Will-ShaoHua — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/model/CodeWhispererModel.kt (hunk 1 above)
Comment (andrewyuq) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/model/CodeWhispererModel.kt:138]: yes, definitely, thanks for catching this.

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt (hunk 2 above)
Comment (andrewyuq) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:907]: I belive in kotlin something like this means it's abstract? Not totally sure, but all the existing subscribers work like this.

Comment (Will-ShaoHua) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:907]: nah i think to make it abstract `fun onEnter()` is sufficient and we don't need `{}`

Comment (Will-ShaoHua) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt:907]: unless we need an implementation of `CodeWhispererIntelliSenseOnHoverListener ` which does nothing

File: plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt [hunk 3]
```diff
@@ -484,6 +501,20 @@ class CodeWhispererPopupManager {
...
         }
+
+        val suggestionHoverEnterListener: EditorMouseMotionListener = object : EditorMouseMotionListener {
+            override fun mouseMoved(e: EditorMouseEvent) {
+                if (e.inlay != null) {
+                    showPopup(sessionContext, force = true)
+                } else {
+                    sessionContext.project.messageBus.syncPublisher(
+                        CodeWhispererService.CODEWHISPERER_INTELLISENSE_POPUP_ON_HOVER,
+                    ).onEnter()
```
Comment (rli) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt:512]: if we already have the `sessionContext`, why do we need the message bus?

Comment (andrewyuq) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt:512]: don't need it here

**Pull request merged**